client/lib/js/react/components/asset_detail.js

```diff
@@ -23,27 +22,40 @@ const AssetDetail = React.createClass({
 
     getAssetContent() {
         const { asset, assetContent } = this.props;
-        if (!assetContent) {
-            return asset.transaction.data ? asset.transaction.data.payload.content : '-';
-        } else {
+
+        if (assetContent) {
             return assetContent;
         }
+
+        const { data: { payload: { content } } = {} } = asset.transaction;
```

Comment:
Unfortunately this isn't going to work. If we can assume `asset.transaction` exists (declare it as such in the propTypes with a `shape` then), then we'd still add a default object on the payload:

```js
const { data: { payload: { content } = {} } = {} } = asset.transaction;
```

The reason I don't like this as much though, is that if `asset.transaction` doesn't exist, you'll be creating two objects just to get back null anyway. Chaining `&&` avoids this.